hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java

```diff
@@ -167,13 +167,22 @@ protected void registerColumnTypes(TypeContributions typeContributions, ServiceR
 		ddlTypeRegistry.addDescriptor( new DdlTypeImpl( SQLXML, "xml", this ) );
 
 		if ( getDB2Version().isBefore( 11 ) ) {
-			// should use 'binary' since version 11
 			ddlTypeRegistry.addDescriptor(
 					CapacityDependentDdlType.builder( BINARY, "varchar($l) for bit data", this )
 							.withTypeCapacity( 254, "char($l) for bit data" )
 							.build()
 			);
 		}
+		else {
+			// should use 'binary' since version 11
+			ddlTypeRegistry.addDescriptor(
+					CapacityDependentDdlType.builder( BINARY, columnType( BINARY ), this )
+							.withTypeCapacity( 255, columnType( BINARY ) )
+							.withTypeCapacity( getMaxVarbinaryLength(), columnType( VARBINARY ) )
+							.withTypeCapacity( 2_147_483_647, columnType( BLOB ) )
+							.build()
+			);
+		}
```

Comment:
Where does this comes from?

Reply:
> Where does this comes from?

From my 11.1 DB2 version changes, not sure why i left it in, removing ...